packages/blade/src/components/TopNav/_decisions/decisions.md

```diff
@@ -264,6 +473,298 @@ const Dashboard = () => {
+                description="Some product description"
+              />
+            </MenuItem>
+            <MenuItem>
+              <ExploreItem
+                icon={RazorXIcon}
+                title="RazorX"
+                description="Some product description"
+              />
+            </MenuItem>
+          </MenuOverlay>
+        </Menu>
+      ),
+    },
+  ]}
+/>
+```
+
+
+**Alternative Approach: JSX with MetaData:**
+
+Another approach where we can pass a new overflowMetadata prop to TabNavItem components and use that to render the overflowingItems later on.
+
+**Pros:**
+1. Familiar compound like API & less verbose
+2. Arguably less confusing than other alternative approaches
+
+**Cons:**
+1. Odd indirection & data flow, where the overflowMetadata gets indirectly passed to the overflowingItems
+2. Introduces 1 extra component TabNavItems which wraps all other TabNavItemLink
+
+```jsx
+const { overflowingItems, setOverflowingItems } = React.useState([]);
+
+<TabNav>
+  <TabNavItems
+    onOverflow={(items) => {
+      setOverflowingItems(items);
+    }}
+  >
+    <TabNavItemLink icon={HomeIcon} accessibilityLabel="Home" href="/home" />
+    <TabNavItemLink title="Payroll" href="/payroll" />
+    <TabNavItemLink
+      title="Payments"
+      href="/payments"
+      icon={MagicCheckoutIcon}
+      // this meta data will be passed to `items` array
+      overflowMetadata={{
+        icon: MagicCheckoutIcon,
+        href: '/payment',
+        title: 'Payments',
+        description: 'Some product description',
+      }}
+    />
+  </TabNavItems>
+  <Menu>
+    <TabNavItemLink title="More" trailing={<ChevronDownIcon />} />
+    <MenuOverlay>
+      // render the overflowing items
+      {overflowingItems.map((item) => {
+        return (
+          <MenuItem key={item.title}>
+            <ExploreItem icon={item.icon} title={item.title} description={item.description} />
+          </MenuItem>
+        );
+      })}
+      <MenuItem>
+        <ExploreItem
+          icon={MagicCheckoutIcon}
+          title="Magic Checkout"
+          description="Some product description"
+        />
+      </MenuItem>
+      <MenuItem>
+        <ExploreItem icon={RazorXIcon} title="RazorX" description="Some product description" />
+      </MenuItem>
+    </MenuOverlay>
+  </Menu>
+</TabNav>
+```
+
+**Alternative Approach: JSX as Data:**
+
+Another approach is this, where we treat the JSX as the Data.
+This is similar to Dropdown where on initial render we loop over all the JSX elements build a list and store it and from that list we render.
+
+**Not leaning towards this API mainly because:**
+1. JSX as Data gets very [complicated](https://github.com/razorpay/blade/blob/2e4b2cb309cdf3e87982cda9252e9ce97747c475/packages/blade/src/components/ActionList/actionListUtils.ts#L137-L159) as seen on Dropdown, and most of the time breaks composition, like it looks like the component is flexible and composable but in reality it isn’t.
```

Comment:
TLDR; 

In dropdown the problem is we do composition via React.Children.map & check for the item's componentIds and if we don't find it we throw error, problem with this is it sort of breaks composability where now if you wrap the Dropdown items in any other wrapper component it will stop working.